pkg/utils/env/openv.go

```diff
@@ -0,0 +1,60 @@
+package env
+
+import (
+	"os"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"k8s.io/client-go/rest"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	v1 "k8s.io/apimachinery/pkg/runtime/schema"
+	serializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+var log = logf.Log.WithName("env")
+
+func DetectOpenshift() (bool, error) {
+
+	log.Info("Detect if openshift is running")
+
+	value, ok := os.LookupEnv("OPERATOR_OPENSHIFT")
+	if ok {
+		log.Info("Set by env-var 'OPERATOR_OPENSHIFT': " + value)
+		return strings.ToLower(value) == "true", nil
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.Error(err, "Error getting config: %v")
+		return false, err
+	}
+
+	groupName := "route.openshift.io"
+	gv := v1.GroupVersion{Group: groupName, Version: "v1"}
+	cfg.APIPath = "/apis"
+
+	scheme := runtime.NewScheme()
+	codecs := serializer.NewCodecFactory(scheme)
+
+	cfg.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: codecs}
+
+	if cfg.UserAgent == "" {
+		cfg.UserAgent = rest.DefaultKubernetesUserAgent()
+	}
+
+	cfg.GroupVersion = &gv
+
+	client, err := rest.RESTClientFor(cfg)
+
+	if err != nil {
+		log.Error(err, "Error getting client: %v")
+		return false, err
+	}
+
+	_, err = client.Get().DoRaw()
+
+	return err == nil, nil
```

Comment:
We also need the version, at least ocp 3 vs 4, but maybe also the exact version. Not sure if that's best in this function or a different one.

Reply:
will look into this and will open a new PR.

Reply:
@redhatHameed No need to open a new PR, you just just push changes to this PR

Reply:
pushed the changes and writing the version details in logs as below

{"level":"info","ts":1558035029.7404926,"logger":"env","msg":"Detect if openshift is running"}
{"level":"info","ts":1558035030.041868,"logger":"env","msg":"OpenShift Version is 3.11+"}
{"level":"info","ts":1558035030.101881,"logger":"controller_activemqartemis","msg":"evnironment is openshift"}

Reply:
I assume these log statements will be in debug verbosity... but what's the plan? Is this commit just to test things out?